### PR TITLE
Rounded border

### PR DESCRIPTION
**Pull request by yshui**

### CODE REVIEW

**Comment by yshui** [comment]
This is part of #229

**Comment by yshui** [comment]
cc @ibhagwan

File: src/win.c [hunk 1]
```diff
@@ -1085,6 +1085,11 @@ static void win_determine_rounded_corners(session_t *ps, struct managed_win *w)
 	} else {
 		w->corner_radius = ps->o.corner_radius;
 		log_debug("Rounding corners for window %#010x", w->base.id);
+		// HACK: we reset this so we can query the color once
+		// we query the color in glx_round_corners_dst0 using glReadPixels
+		// w->border_col = { -1., -1, -1, -1 };
```
Comment (tryone144) [inline src/win.c:1090]: Just for consistency, the function is now called `glx_round_corners_dst()` and the first value for `border_col` is written with a trailing dot.

**Review by tryone144 — APPROVED** [review]
This looks okay to me.
Unfortunately, I can't really test it locally at the moment as I only have i3's fake borders at hand and my test-system doesn't cooperate.

File: src/win.c (hunk 1 above)
Comment (yshui) [reply src/win.c:1090]: Ah, thanks for pointing it out!

**Comment by yshui** [comment]
@tryone144 I think the approach should work with "fake" borders

**Comment by owl4ce** [comment]
Hi, is it possible to turn off borders on this window?

2021-05-30-095805_379x397_scrot

2021-05-30-095821_833x382_scrot

This only happens when turning on rounded corners with the `glx` backend. I use OpenboxWM.
2021-05-30-100850_266x216_scrot

**Comment by ibhagwan** [comment]
> This only happens when turning on rounded corners with the `glx` backend. I use OpenboxWM.

That’s because rounded borders are only supported on `glx`.

You can turn it off globally with the `round-borders = 0;` option or individually with the `round-borders-exclude` option, use `xprop` for the windows you’d like it turned off and refer to [picom.sample.conf](https://github.com/ibhagwan/picom/blob/next-rebase/picom.sample.conf) if you’re unclear how to use the option.

**Comment by owl4ce** [comment]
> > This only happens when turning on rounded corners with the `glx` backend. I use OpenboxWM.
> 
> That’s because rounded borders are only supported on `glx`.
> 
> You can turn it off globally with the `round-borders = 0;` option or individually with the `round-borders-exclude` option, use `xprop` for the windows you’d like it turned off and refer to [picom.sample.conf](https://github.com/ibhagwan/picom/blob/next-rebase/picom.sample.conf) if you’re unclear how to use the option.

Using all border options doesn't work to disable. It only happens in Openbox's own drawn window and can't use `xprop` because of conflict when grabbing the cursor.

**Comment by ibhagwan** [comment]
> > > This only happens when turning on rounded corners with the `glx` backend. I use OpenboxWM.
> > 
> > 
> > That’s because rounded borders are only supported on `glx`.
> > You can turn it off globally with the `round-borders = 0;` option or individually with the `round-borders-exclude` option, use `xprop` for the windows you’d like it turned off and refer to [picom.sample.conf](https://github.com/ibhagwan/picom/blob/next-rebase/picom.sample.conf) if you’re unclear how to use the option.
> 
> Using all border options doesn't work to disable. It only happens in Openbox's own drawn window and can't use `xprop` because of conflict when grabbing the cursor.
> 
> **EDIT:** That includes this rule `"! name~=''"`

I took a quick glance at the code, I don’t think the border exclude options were implemented here yet, if you’d like to give these options a try use [my fork instead](https://github.com/ibhagwan/picom).

**Comment by owl4ce** [comment]
Solved for temporary, back using picom from this commit.
https://github.com/yshui/picom/commit/9cb552ecd91ec644bf6fcb558ddd44fda5b4f7d9

**Comment by owl4ce** [comment]
Solved for temporary, back using picom from this commit.
https://github.com/yshui/picom/commit/9cb552ecd91ec644bf6fcb558ddd44fda5b4f7d9

**Pull request merged**